libraries/rush-lib/src/logic/base/BaseInstallManager.ts

```diff
@@ -597,6 +598,30 @@ ${gitLfsHookHandling}
         args.push('--strict-peer-dependencies');
       }
 
+      /*
+        If user set resolution-mode in pnpm-config.json only, use the value in pnpm-config.json
+        If user set resolution-mode in pnpm-config.json and .npmrc, use the value in pnpm-config.json
+        If user set resolution-mode in .npmrc only, do nothing, let pnpm handle it
+        If user does not set resolution-mode in pnpm-config.json and .npmrc, rush will default it to "highest"
+      */
+      const isResolutionModeInNpmrc: boolean = isVariableSetInNpmrcFile(
+        this.rushConfiguration.commonRushConfigFolder,
+        'resolution-mode'
+      );
+      if (this.rushConfiguration.pnpmOptions.resolutionMode) {
+        if (isResolutionModeInNpmrc) {
+          console.log(
+            colors.yellow(
+              `Warning: PNPM's resolution-mode is specified in both .npmrc and pnpm-config.json. The value in pnpm-config.json will take precedence.`
+            )
+          );
```

Comment:
```suggestion
          this._terminal.writeWarningLine(
            `Warning: PNPM's resolution-mode is specified in both .npmrc and pnpm-config.json. ` +
              `The value in pnpm-config.json will take precedence.`
          );
```

Reply:
We're trying to move everything to `Terminal`. `Terminal.writeWarningLine` by default prints in Yellow.